api/service.go

```diff
@@ -893,91 +930,60 @@ func (s *service) CatFile(req *pb.CatFileRequest, server pb.API_CatFileServer) e
 	return nil
 }
 
-// RemoveFile handles a remove file request.
-func (s *service) RemoveFile(ctx context.Context, req *pb.RemoveFileRequest) (*pb.RemoveFileReply, error) {
-	log.Debugf("received remove file request")
+// RemoveBucketPath handles a remove bucket path request.
+func (s *service) RemoveBucketPath(
+	ctx context.Context,
+	req *pb.RemoveBucketPathRequest,
+) (*pb.RemoveBucketPathReply, error) {
+	log.Debugf("received remove bucket path request")
 
 	scope, ok := ctx.Value(reqKey("scope")).(string)
 	if !ok {
 		log.Fatal("scope required")
 	}
-	folderName, fileName := parsePath(req.Path)
-	folder, err := s.getFolderWithScope(ctx, folderName, scope)
+	buckName, fileName := parsePath(req.Path)
+	buck, err := s.getBucketWithScope(ctx, buckName, scope)
 	if err != nil {
 		return nil, err
 	}
-	folderPath := path.New(folder.Path)
 
-	dirpth, err := s.ipfsClient.Object().RmLink(ctx, folderPath, fileName)
+	buckPath := path.New(buck.Path)
+	dirpth, err := s.ipfsClient.Object().RmLink(ctx, buckPath, fileName)
 	if err != nil {
 		return nil, err
 	}
-	if err = s.ipfsClient.Pin().Update(ctx, folderPath, dirpth); err != nil {
-		return nil, err
-	}
-	if err = s.collections.Folders.UpdatePath(ctx, folder, dirpth); err != nil {
-		return nil, err
-	}
-
-	return &pb.RemoveFileReply{}, nil
-}
 
-// getTeamForUser returns a team if the user is authorized.
-func (s *service) getTeamForUser(ctx context.Context, teamID string, user *c.User) (*c.Team, error) {
-	team, err := s.collections.Teams.Get(ctx, teamID)
+	links, err := s.ipfsClient.Unixfs().Ls(ctx, dirpth)
 	if err != nil {
 		return nil, err
 	}
-	if team == nil {
-		return nil, status.Error(codes.NotFound, "Team not found")
-	}
-	if !s.collections.Users.HasTeam(user, team.ID) {
-		return nil, status.Error(codes.PermissionDenied, "User is not a team member")
-	}
-	return team, nil
-}
 
-// getProjectForScope returns a project if the scope is authorized.
-func (s *service) getProjectForScope(ctx context.Context, projID, scope string) (*c.Project, error) {
-	proj, err := s.collections.Projects.Get(ctx, projID)
-	if err != nil {
-		return nil, err
+	var cnt int
+	for l := range links {
```

Comment:
Uhm, I wonder how the creators of the API thought about a closed channel because there was some error. Seems that that case can't be distinguished. I don't see some `.Err()` (similar to how it's done in other places).
Anyway, not much we can do. But mentioning it because I may be missing something.

Reply:
Yeah, strange. The API used to return a list of links. Not sure why they changed to channel.